### PR TITLE
WT-4798 Improve eviction efficiency with many active tables.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

**Comment by michaelcahill** [comment]
For discussion today: in my testing with 2000 MongoDB collections, this change resolves inefficiency in eviction.  It may have unintended consequences for other scenarios: my testing so far has been focused on one use case.

**Comment by michaelcahill** [comment]
@sulabhM, @ddanderson after discussion this afternoon, I pushed a change in 0faba83 that I hope restores some of the fairness that the original change could remove (by eliminating the factor of 10 in `QUEUE_FILLS_PER_PASS`).

I also changed `__evict_walk_target` to calculate relative to the total queue size rather than the remaining slots, which I think addresses WT-4802.  It would be fine to split that part out, but I thought it made sense to test the changes together.

**Comment by michaelcahill** [comment]
There's more to do here: `test_wt4333_handle_locks` is still failing in automated testing.  For some reason, the changes in this branch mean we're failing to make progress in that (admittedly odd) case where there are 1000 trees, each with 19 dirty leaf pages in cache...

**Comment by michaelcahill** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
autobuild: retest this please

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -1755,8 +1733,7 @@ __evict_walk_target(WT_SESSION_IMPL *session, u_int max_entries)
 	 * small trees, so round to a whole number of slots (zero for small
```
Comment (sulabhM) [inline src/evict/evict_lru.c]: We  do not need this comment anymore.

**Comment by sulabhM** [comment]
WT-4878 merged the changes to target calculations.
WT-4802 will address the rest of the improvements and enable random dhandle selection. I have made reference to this pull request in WT-4802.

No further work needs to be done as part of this pull request, so I am closing it for now.

**Pull request closed without merging**